### ISSUE TITLE
PDF.JS and IE10 performance issue

### ISSUE THREAD

**Issue by jerpoubelle** [mozilla/pdf.js#5408]
Hi,

I am having performance issues on IE10 compared to FF or Chrome.

For example, if i load this PDF in a PDF.js viewer :
http://www.lifeway.com/lwc/files/lwcf_crd_mgz_essentialconnectionsample_pdf.pdf

the pdf is loaded almoast instantaneously in FF or Chrome where it takes about 10seconds to load in IE10.

i foudn a reference to this kind of issue but the topic has been closed :
https://github.com/mozilla/pdf.js/issues/2873

I'm using pdfjs-1.0.473 and windows 7 / IE10.

Thanks in advance

**Comment by matthiasg** [comment]
When opening http://mozilla.github.io/pdf.js/web/viewer.html with IE11 (11.0.10011.0) I only get the toolbar.

Same with Spartan as in Win10 10049. Only the static top toolbar is rendered. Nothing else happens. Same page renders fast and fine on Chrome 41.

**Comment by FMWS** [comment]
I found the time it takes for the promise returned by PDFPageView.getTextContent to resolve to be a lot higher in IE11 compared to Firefox on Windows 8.1. I added lines like `console.time("getTextContent" + self.id);` and `console.timeEnd("getTextContent" + self.id);` at around line 4010 to the viewer example. In IE11 the time it took for promises to resolve for pages in "Trace-based Just-in-Time Type Specialization for DynamicLanguages" was up to 60 times that of Firefox (up to 4 seconds in IE). In other documents and using a different viewer built around pdf.js (which I can't share, sorry) it's even worse, with some pages taking more than 2 minutes (vs 7 seconds in FF) to resolve the getTextContent promise.
